exchanges/binance/binance_wrapper.go

```diff
@@ -836,47 +1262,205 @@ func (b *Binance) GetOrderHistory(req *order.GetOrdersRequest) ([]order.Detail,
+				if req.EndTicks.After(req.StartTicks) {
+					if time.Since(req.StartTicks) > time.Hour*24*30 {
+						return nil, fmt.Errorf("can only fetch orders 7 days out")
+					}
+					orderHistory, err := b.GetAllFuturesOrders(req.Pairs[i], "", req.StartTicks, req.EndTicks, 0, 0)
+					if err != nil {
+						return nil, err
+					}
+					for y := range orderHistory {
+						var feeBuilder exchange.FeeBuilder
+						feeBuilder.Amount = orderHistory[y].ExecutedQty
+						feeBuilder.PurchasePrice = orderHistory[y].AvgPrice
+						feeBuilder.Pair = req.Pairs[i]
+						fee, err := b.GetFee(&feeBuilder)
+						if err != nil {
+							return orders, err
+						}
+						orderVars := compatibleOrderVars(orderHistory[y].Side, orderHistory[y].Status, orderHistory[y].OrderType)
+						orders = append(orders, order.Detail{
+							Price:           orderHistory[y].Price,
+							Amount:          orderHistory[y].OrigQty,
+							ExecutedAmount:  orderHistory[y].ExecutedQty,
+							RemainingAmount: orderHistory[y].OrigQty - orderHistory[y].ExecutedQty,
+							Fee:             fee,
+							Exchange:        b.Name,
+							ID:              strconv.FormatInt(orderHistory[y].OrderID, 10),
+							ClientOrderID:   orderHistory[y].ClientOrderID,
+							Type:            orderVars.OrderType,
+							Side:            orderVars.Side,
+							Status:          orderVars.Status,
+							Pair:            req.Pairs[i],
+							AssetType:       asset.CoinMarginedFutures,
+						})
+					}
+				}
+			case req.OrderID != "" && req.StartTicks.IsZero() && req.EndTicks.IsZero():
+				fromID, err := strconv.ParseInt(req.OrderID, 10, 64)
+				if err != nil {
+					return nil, err
+				}
+				orderHistory, err := b.GetAllFuturesOrders(req.Pairs[i], "", time.Time{}, time.Time{}, fromID, 0)
+				if err != nil {
+					return nil, err
+				}
+				for y := range orderHistory {
+					var feeBuilder exchange.FeeBuilder
+					feeBuilder.Amount = orderHistory[y].ExecutedQty
+					feeBuilder.PurchasePrice = orderHistory[y].AvgPrice
+					feeBuilder.Pair = req.Pairs[i]
+					fee, err := b.GetFee(&feeBuilder)
+					if err != nil {
+						return orders, err
+					}
+					orderVars := compatibleOrderVars(orderHistory[y].Side, orderHistory[y].Status, orderHistory[y].OrderType)
+					orders = append(orders, order.Detail{
+						Price:           orderHistory[y].Price,
+						Amount:          orderHistory[y].OrigQty,
+						ExecutedAmount:  orderHistory[y].ExecutedQty,
+						RemainingAmount: orderHistory[y].OrigQty - orderHistory[y].ExecutedQty,
+						Fee:             fee,
+						Exchange:        b.Name,
+						ID:              strconv.FormatInt(orderHistory[y].OrderID, 10),
+						ClientOrderID:   orderHistory[y].ClientOrderID,
+						Type:            orderVars.OrderType,
+						Side:            orderVars.Side,
+						Status:          orderVars.Status,
+						Pair:            req.Pairs[i],
+						AssetType:       asset.CoinMarginedFutures,
+					})
+				}
```

Comment:
I'm not sure why this needs two cases. The only difference is with the `time.Time` checks and `fromID`. This could be combined with just some extra checks